### DISCUSSION TITLE
USA BlueLink Access Limit Reduced?

### DISCUSSION THREAD

**Discussion by gyveri** [ZuinigeRijder/hyundai_kia_connect_monitor#61] [Q&A]
The U.S. Bluelink Access Limit appears to be less than the 30 per day stated in ReadMe. I have run run_monitor_once running once every hour on the hour for the past several days. Every day, monitor.csv shows one record per hour from 02:00 to 21:00, with a final entry at 21:03. Every day, the run_monitor_once.log includes four error messages that read "hyundai_kia_connect_api - get vehicle location rate limit exceeded".
I realize that I could avoid this issue by running run_monitor_once less than 24 times per day, but I charge at night and sometimes make per-dawn trips, so the choice the hours to exclude would be difficult. 
Has the Limit actually changed, or is something I don't know about accessing Bluelink more than once per hour (I don't access Bluelink using my computer or my phone)?
If the Limit has changed, can I still get 24-hour coverage by accessing it once every 2 hours and, during each access, requesting data for each of the previous 2 hours? It's not clear to me what the "per action" limit refers to.

**Comment by acmay** [comment]
Hyundai likes to blame 3rd party apps for the 12v battery drain like this article.
https://insideevs.com/news/658984/hyundai-finds-causes-for-ioniq-5-12v-battery-drain-reveals-fixes/

So they maybe working to stop the apps rather than fix their software problems.

**Comment by ZuinigeRijder** [comment]
> If the Limit has changed, can I still get 24-hour coverage by accessing it once every 2 hours and, during each access, requesting data for each of the previous 2 hours? It's not clear to me what the "per action" limit refers to.

@gyveri The monitor only asks for the cached values stored on the server. If you ask for the information at 00:00, it will get the latest server values. When you do not drive at night, but charge at night and you want to capture the end SOC% before going to drive, you can run monitor from the earliest time you expect you will leave in the morning. So skipping e.g. 0:00 AM till 4:00 AM. Another approach or combined approach is running monitor each 1.5 hour or 2 hours as you mentioned.

There is always a chance that you miss the exact SOC% when leaving in the morning, depending when the charging is ended. The IONIQ 5 is only sending new updates to the server when the car is turned off, the charging has ended  (sometimes also 10% increase increments) and when you do a refresh in the Bluelink App to get the latest values of the car. Note that monitor never awakes the car by forcing a refresh.

You are sure you do not use or registered once other 3-part apps for access to Bluelink, e.g. for cheaper charging rates? I know this is the case here in Europe, just by registering the e.g. Hyundai Charging App for load balancing.

Another idea would be to trigger a run of monitor on your server to run it also at request, but then you must think about how to do this and not to forget to run it, when you leave e.g. very early in the morning. Or trigger it once when the charge has ended. But then you should make something on top of hyundai_kia_connect_monitor to trigger this.

**Reply by gyveri** [reply]
I'm trying to understand more about "The monitor only asks for the cached values stored on the server." How much of the data that is stored on the server can the monitor ask for? Can only the most-recent data upload from the car be asked for, or can one or more previous uploads also be asked for? If one were to run monitor every 2 hours, would the monitor automatically ask for and receive all of the uploads that have not previously been asked for? Or would it have to be modified to ask for more than the most-recent upload. I've tried to figure this out for myself, but, although I've written several python scripts to gather data from my weather station, my solar panels, my electricity meter monitor and PowerWall battery, your scripts are way above my python skill level.

**Comment by ZuinigeRijder** [comment]
It is simple, the information in monitor.csv, there is no history there, except for the latest values available on the server. My tool only adds entries when the latest entry is different from the received information.

monitor.dailystats.csv and monitor.tripinfo.csv has history till about 3 months ago. However, not in all regions dailystats and tripinfo is available. It is available in Europe, China and Australia, but not (yet) in USA and Canada. Maybe this information is available (if this information is available in the App), but then someone should try to sniff the calls from the App to the server and add it to [hyundai_kia_connect_api.](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api)
